test/DebuggerTesting/OpenDebug/Events/OutputEvent.cs

```diff
@@ -0,0 +1,43 @@
+﻿// // Copyright (c) Microsoft. All rights reserved.
+// // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace DebuggerTesting.OpenDebug.Events
+{
+    public enum CategoryValue
+    {
+        Console = 0,
+        Stdout = 1,
+        Stderr = 2,
+        Telemetry = 3,
+        Unknown = Int32.MaxValue
+    }
+
+    public sealed class OutputEventValue : EventValue
+    {
+        public sealed class Body
+        {
+            public string output;
+
+            public string category;
+        }
+
+        public Body body = new Body();
+    }
+
+    public sealed class OutputEvent : Event<OutputEventValue>
+    {
+        public OutputEvent(string text, CategoryValue category) : base("output")
+        {
+            this.ExpectedResponse.body.category = GeCategory(category);
+            this.ExpectedResponse.body.output = text;
```

Comment:
Is there a good way to make this a `Contains` check instead of a strict `string.equals`?

Reply:
Another thing is, do I need a way to check the next `n` OutputEvents since the first one won't be the LogMessage?

Reply:
I am not sure if the Cpp tests stuff works differently, but I think this is just using the DAR stuff for comparing events. If so, all strings are treated as regular expressions.

Reply:
```
 Expected = {"body":{"output":"Log Message(\r\n|\r|\n)","category":"console"},"event":"output"}
 Actual Responses =
 1. {"event":"breakpoint","body":{"reason":"changed","breakpoint":{"id":2,"verified":true,"line":52,"BoundBreakpoints":[]}},"seq":40,"type":"event"}
 2. {"event":"output","body":{"category":"stdout","output":"Calling recursive function.\n"},"seq":42,"type":"event"}
 3. {"event":"output","body":{"category":"stdout","output":"\n"},"seq":43,"type":"event"}
 4. {"event":"output","body":{"category":"stdout","output":"Breakpoint 3, Calling::CoreRun (this=0x7fffffffda50) at /home/runner/work/MIEngine/MIEngine/bin/DebugAdapterProtocolTests/Debug/CppTests/debuggees/instance/kitchensink-3/src/calling.cpp:52\n"},"seq":44,"type":"event"}
 5. {"event":"output","body":{"category":"stdout","output":"52\t    this->Log(\"Calling variable arg function.\");\n"},"seq":45,"type":"event"}
 6. {"event":"output","body":{"category":"console","output":"Log Message\n"},"seq":46,"type":"event"}
 ```

I'm a bit confused why DAR is unhappy on linux even though the event is being shown. My thoughts were "do I need a way to check the next n OutputEvents since the first one won't be the LogMessage?" but it seems to work correctly on windows and macOS.

Reply:
Does `"\\bLog Message\\b"` work? I am not sure why `\n` doesn't work. The only guess I have is that it is connected with the fact that we aren't in ["Single-line mode"](https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-options#single-line-mode) and you are using `\n` instead of `\\n`. But this is a total guess.

Reply:
I tried running the snippet on Linux, and I got the result we expect, so no idea. You might just have to debug it.

Reply:
Did you try this in WSL or a Linux VM? I was able to have it succeeded in WSL locally, but the test machine runs fail.

My next steps is running in a VM.

Reply:
I was also using WSL. Did you try `"^Log Message\\b"`?

Reply:
Both `"Log Message(\r\n|\r|\n)"` and `"^Log Message\\b"` succeeded in WSL.

Reply:
Yep. Reproduces in a Linux VM.